lib/src/widget/nav_bar/utils/observer.dart

```diff
@@ -0,0 +1,30 @@
+import 'package:flutter/material.dart';
+import 'package:xayn_design/src/widget/nav_bar/widget/nav_bar_container.dart';
+
+class NavBarObserver extends NavigatorObserver {
+  @override
+  void didPop(Route route, Route? previousRoute) {
+    _resetNavBarConfig(true);
```

Comment:
I hope that observer works with the Navigator2.0

Reply:
discussed. will figure out (update this one) later